core/src/main/java/org/apache/accumulo/core/clientImpl/ReplicationOperationsImpl.java

```diff
@@ -113,8 +114,20 @@ public void drain(final String tableName, final Set<String> wals)
   protected boolean getMasterDrain(final TInfo tinfo, final TCredentials rpcCreds,
       final String tableName, final Set<String> wals)
       throws AccumuloException, AccumuloSecurityException, TableNotFoundException {
-    return MasterClient.execute(context,
-        client -> client.drainReplicationTable(tinfo, rpcCreds, tableName, wals));
+    boolean result;
+    try {
+      result = MasterClient.execute(context,
+          client -> client.drainReplicationTable(tinfo, rpcCreds, tableName, wals));
+    } catch (ThriftTableOperationException e) {
+      switch (e.getType()) {
+        case NAMESPACE_NOTFOUND:
```

Comment:
Should this case throw NamespaceNotFoundException instead? Or at least, TNFE caused by NNFE?

Reply:
Not sure.  I was thinking from a user perspective, they probably don't care to catch a TNFE AND NNFE.  I had thought about changing it to also throw NamespaceNotFound as well but I can't think of a situation where you would handle one differently then the other.

Reply:
Some of the previous code did:

```java
throw new TableNotFoundException(e.getTableName(), new NamespaceNotFoundException(e))
```

This explained that the table didn't exist, but the underlying cause was that the namespace for the table didn't exist.

Reply:
I guess that makes sense.  The whole point of this PR was to refactor this wrapping.  I will make this PR just update the javadoc.

Reply:
Oh, I see. Yeah, I guess it's ugly, but I'm not sure what's better right now.